ngraph_bridge/ngraph_encapsulate_op.cc

```diff
@@ -53,1019 +57,617 @@ namespace ng = ngraph;
-  explicit NGraphEncapsulateOp(OpKernelConstruction* ctx)
-      : OpKernel(ctx),
-        m_graph(OpRegistry::Global()),
-        m_freshness_tracker(nullptr) {
-    my_instance_id = s_instance_count;
-    s_instance_count++;
-
-    std::ostringstream oss;
-    oss << "Encapsulate_" << my_instance_id << ": " << name();
-    ngraph::Event event(oss.str(), name(), "");
-
-    NGRAPH_VLOG(1) << "NGraphEncapsulateOp: " << my_instance_id
-                   << " Name: " << name();
-
-    GraphDef* graph_def;
-
-    OP_REQUIRES_OK(ctx, ctx->GetAttr<int>("ngraph_cluster", &m_ngraph_cluster));
-    graph_def = NGraphClusterManager::GetClusterGraph(m_ngraph_cluster);
-
-    if (graph_def == nullptr) {
-      string flib_key = "ngraph_cluster_" + to_string(m_ngraph_cluster);
-      // Read graphdef from function library
-      const FunctionLibraryDefinition flib =
-          *ctx->function_library()->GetFunctionLibraryDefinition();
-      const FunctionDef* fdef = flib.Find(flib_key);
-      OP_REQUIRES(
-          ctx, fdef != nullptr,
-          errors::Internal("Did not find graphdef for encapsulate ", flib_key,
-                           " in NGraphClusterManager or function library"));
-      // TODO: how to convert from functiondef to graphdef. Anything easier?
-      std::unique_ptr<FunctionBody> fnbody;
-      const auto get_func_sig = [&flib](const string& op, const OpDef** sig) {
-        return flib.LookUpOpDef(op, sig);
-      };
-      OP_REQUIRES_OK(ctx, FunctionDefToBodyHelper(*fdef, {}, &flib,
-                                                  get_func_sig, &fnbody));
-      CopyGraph(*fnbody->graph, &m_graph);
-    } else {
-      GraphConstructorOptions opts;
-      opts.allow_internal_ops = true;
-      OP_REQUIRES_OK(ctx, ConvertGraphDefToGraph(opts, *graph_def, &m_graph));
-    }
-    OP_REQUIRES_OK(ctx, ctx->GetAttr("ngraph_graph_id", &m_graph_id));
-    //
-    // Initialize the "m_input_is_static" vector as follows:
-    // (1) create m_input_is_static with n+1 elements, where n is the max arg
-    //     index
-    // (2) for each _Arg node n, set m_input_is_static[n.index] to true if n
-    //     is driving any static input; else set it to false.
-    //
-
-    // Create the vector.
-    int32 max_arg_index = -1;
-    std::vector<const Node*> arg_nodes;
-
-    for (auto node : m_graph.nodes()) {
-      if (node->type_string() == "_Arg") {
-        arg_nodes.push_back(node);
-
-        int32 index;
-        OP_REQUIRES_OK(ctx, GetNodeAttr(node->attrs(), "index", &index));
-        if (index > max_arg_index) max_arg_index = index;
-      }
-    }
+//---------------------------------------------------------------------------
+//  NGraphEncapsulateOp::ctor
+//---------------------------------------------------------------------------
+NGraphEncapsulateOp::NGraphEncapsulateOp(OpKernelConstruction* ctx)
+    : OpKernel(ctx) {
+  ng_encap_impl = new NGraphEncapsulateImpl(name());
```

Comment:
No need for new as explained in the .h file. But also - I don't see you delete the object when done (will result in memory leak).